pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -238,6 +238,36 @@ func getPipelineRunUpdates(t *testing.T, actions []ktesting.Action) []*v1beta1.P
 }
 
 func TestReconcile(t *testing.T) {
+	testCases := []struct {
```

Comment:
I see a lot of tests with these same test cases. They could probably be combined into one, e.g. 
```
testSuites := []func(*testing.T){testReconcileWithEmbeddedStatus, testReconcileWithCustomTask...}
for _, testCase := range testCases {
   for _, testClass := range testSuites {
      t.Run(tc.Name, testClass(t, tc.embeddedStatus))
   }
}

Reply:
Good question. I’d lean against this for clarity and ease of running individual test cases in an IDE.

Reply:
at the very least maybe the test cases struct can be defined only once?

Reply:
I considered that, but at least in Goland/IntelliJ, if you don't have the test case structs defined in the test function, you can't launch them individually.